### ISSUE TITLE
[studio][studio-ui] Blob Store: Handle case when blob item is missing but present site git/db

### ISSUE THREAD

**Issue by yacdaniel** [craftercms/craftercms#5972]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When a user tries to interact with a blob item which is missing but exists in site repo/db, some of Studio API's and the Studio UI are not handling this case correctly.

Related to https://github.com/craftercms/craftercms/issues/5949

### Steps to reproduce

Steps:

1. Configure blob store in a site, do a `Publish Entire Project`
2. Upload a blob through Studio so the blob is stored in S3 and the .blob is created in the sandbox repository
3. Delete the file directly from S3
4. Interact with the blob item from Studio UI.

Failing cases:

* Dependencies: 
https://gist.github.com/yacdaniel/baf8522012336c09d0ce3ef0bcb87c85

* Rename:
Fails because calls the API dependencies. 

* Duplicate:
https://gist.github.com/yacdaniel/94616e67025766898975662fe45c741e

* Publish calls the `item_by_path` 
https://gist.github.com/yacdaniel/1071679094b37a9cc4e390fca2440cec

* Copy calls the `sandbox_item_by_path`
The API returns the item as missing
```
{"response":{"code":0,"message":"OK","remedialAction":"","documentationUrl":""},"missingItems":["/static-assets/s3/banquero.png"],"items":[]}

[WARN] 2023-03-03T10:57:40,775 [http-nio-8080-exec-6] [internal.ContentServiceInternalImpl] | Content not found in site 'empty1234' path '/static-assets/s3/banquero.png' 
```

### Relevant log output

```shell
Dependencies API 1

http://localhost:8080/studio/api/1/services/api/1/dependency/get-dependant.json?path=%2Fstatic-assets%2Fs3%2Fbanquero.png&site=empty1234
Request Method: POST
Status Code: 500

Duplicate API 2
Request URL: http://localhost:8080/studio/api/2/content/duplicate
Request Method: POST
Status Code: 500 

Item by path API 2
Request URL: http://localhost:8080/studio/api/2/content/item_by_path?path=%2Fstatic-assets%2Fs3%2Fbanquero.png&preferContent=true&siteId=empty1234
Request Method: GET
```

### Screenshots and/or videos

* Dependencies
![image](https://user-images.githubusercontent.com/6722074/222777406-981e466b-7aba-46d8-b092-df3cea1c4544.png)

* Rename
![image](https://user-images.githubusercontent.com/6722074/222778024-f8646b6b-5a01-4e8d-8358-0ce0d4ce74f4.png)

* Duplicate
![image](https://user-images.githubusercontent.com/6722074/222778606-85f8ee50-91e2-4a8c-a36b-16fd98cef4be.png)

* Publish
![image](https://user-images.githubusercontent.com/6722074/222779358-3306b67c-8aa6-415c-87e6-03383b861a30.png)

* Copy
![image](https://user-images.githubusercontent.com/6722074/222780661-1d05ec41-db77-4dd9-8322-b5280b9c1d61.png)

**Comment by rart** [comment]
@jvega190 please fix the PublishingDialog including removing the `useLogicResource`

**Comment by yacdaniel** [comment]
Verified fixed in 4.0.3-SNAPSHOT

- Dependencies / Rename  returns a 404
![image](https://user-images.githubusercontent.com/6722074/224227580-36414c64-d67a-468e-bdfa-f3a7a3fea8e4.png)

![image](https://user-images.githubusercontent.com/6722074/224227666-2053d3da-8683-4d92-b631-d219b69bf730.png)

- Duplicate returns a 404
![image](https://user-images.githubusercontent.com/6722074/224227908-d589fb05-5037-4a51-bfc6-396224e51496.png)

- Copy UI  prints the missing item exception
![image](https://user-images.githubusercontent.com/6722074/224228184-30ce4907-eda5-47e9-be59-a5349072f2bb.png)

- Publish UI returns the 404 with the new blob store missing content exception
![image](https://user-images.githubusercontent.com/6722074/224228421-48835aa5-3d50-404e-8f75-0a3e6dbd844f.png)
